apps/ejabberd/src/mod_mam_utils.erl

```diff
@@ -668,10 +680,28 @@ form_field(Type, VarName) ->
                     {<<"var">>, VarName}]}.
 
 %% -----------------------------------------------------------------------
+%% Text search tokenization
+
+-spec normalize_search_text(binary() | string() | undefined) -> string() | undefined.
+normalize_search_text(Text) ->
+    normalize_search_text(Text, "%").
+
+-spec normalize_search_text(binary() | string() | undefined, string()) -> string() | undefined.
+normalize_search_text(undefined, _WordSeparator) ->
+    undefined;
+normalize_search_text(Text, WordSeparator) ->
```

Comment:
A word (or more) of an explanation will makes things easier to understand. A simple example may also help.